### PR TITLE
Formally verified OSM

### PR DESCRIPTION
**Pull request by livnev**

### CODE REVIEW

File: src/value.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.6;
+
+contract Value {
```
Comment (rainbreak) [inline src/value.sol:18]: n.b. this has a different storage layout to a DSValue.

**Comment by livnev** [comment]
See here for the 100% spec coverage:
https://dapp.ci/k-osm/f575335dddd663cfb848/

File: src/value.sol (hunk 1 above)
Comment (rainbreak) [reply src/value.sol:18]: see e.g. https://dapp.ci/k-dss/a8c48ed1c8546a39c21a/#DSValue_peek() for DSValue specs.

Comment (rainbreak) [reply src/value.sol:18]: Have updated it to use DSValue, just waiting on the fv build queued here https://buildbot.dapp.ci/#/builders/8

**Comment by rainbreak** [comment]
Updated specs https://dapp.ci/k-osm/9c6e9970bfc9175f55b1/

**Pull request merged**